### PR TITLE
Users can search for specific hotels

### PR DESCRIPTION
**Pull request by mikeymurph77**
@jsteiner, @gabebw

Finally! Thanks to you guys of course.

### CODE REVIEW

File: hotlr/hotel.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Hotel
+  def initialize(row)
+    @row = row
+  end
+
+  def name_of_hotel
```
Comment (gabebw) [inline hotlr/hotel.rb]: `hotel.name_of_hotel` is a little redundant. How about calling this method `name` instead?

File: hotlr/hotel_search.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  def initialize(hotels)
+    @hotels = hotels
+  end
+
+  def search
+    promt_for_hotel_name
+    search_match_hotel
+  end
+
+  def promt_for_hotel_name
```
Comment (gabebw) [inline hotlr/hotel_search.rb]: Typo: `prompt` not `promt`.

File: hotlr/hotel_search.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    puts "What property? > "
+    @user_search = gets.chomp
+  end
+
+  def search_match_hotel
```
Comment (gabebw) [inline hotlr/hotel_search.rb]: This reads kind of oddly - not sure if it's a verb or a noun. How about `find_matching_hotel`?

File: hotlr/hotel_search.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    found_hotel = @hotels.find { |hotel|
```
Comment (gabebw) [inline hotlr/hotel_search.rb]: This is doing 2 things: finding the matching hotel, and outputting it. Can you make it so this method just returns the found hotel, and you output its name elsewhere?

File: hotlr/hotel_search.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      hotel.name_of_hotel == @user_search 
+    } || NullHotel.new
+
+    puts found_hotel.name_of_hotel
+  end
+end
+
+class NullHotel
```
Comment (gabebw) [inline hotlr/hotel_search.rb]: This should be in its own file.

File: hotlr/hotel_search.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
+require "./hotel"
+require "./hotel_reader"
+
+class HotelSearch
+  def initialize(hotels)
+    @hotels = hotels
+  end
+
+  def search
+    promt_for_hotel_name
```
Comment (gabebw) [inline hotlr/hotel_search.rb]: Nice!

**Comment by mikeymurph77** [comment]
@jsteiner & @gabebw 

Is this ready to move on to step 3?

File: hotlr/hotel_search.rb [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      hotel.name == @user_search 
+    } || NullHotel.new
+  end
+
+  def query_result
+    puts @found_hotel.name
+  end
+end
+
+class NullHotel
```
Comment (gabebw) [inline hotlr/hotel_search.rb]: This should be in its own file.

File: hotlr/hotel_search.rb [hunk 8]
```diff
@@ -0,0 +1,41 @@
+require "./hotel"
```
Comment (gabebw) [inline hotlr/hotel_search.rb:1]: `hotel_reader.rb` already requires `hotel`, so you should either have it here OR in `hotel_reader.rb`. It won't break anything, but it's good to be aware of what is requiring what.

I think it's a good idea to require everything in the "main" file that you run at the command line, so this is a good place for it. I'd remove it from `hotel_reader.rb`.

File: hotlr/hotel_reader.rb [hunk 9]
```diff
@@ -1,17 +1,19 @@
 require "csv"
 require "pp"
 
+require "./hotel"
+
 class HotelReader
```
Comment (gabebw) [inline hotlr/hotel_reader.rb:4]: This class looks good!

**Comment by mikeymurph77** [comment]
@jsteiner @gabebw 
Put null_hotel class in it's own file.
All required files stored in hotel_search.rb

**Comment by gabebw** [comment]
:+1: Merge away!

**Comment by mikeymurph77** [comment]
@jsteiner, @gabebw 
I believe I completed step 4.
4. Display information for the hotel that was found

**Comment by gabebw** [comment]
Yes, you did.

**Comment by mikeymurph77** [comment]
@gabebw :+1: ?

File: hotlr/hotel_search.rb [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    query_result
+  end
+
+  def prompt_for_hotel_name
+    puts "What property? > "
+    @user_search = gets.chomp
+  end
+
+  def find_matching_hotel
+    @found_hotel = @hotels.find { |hotel|
```
Comment (gabebw) [inline hotlr/hotel_search.rb]: Can you remove this instance variable?

**Comment by mikeymurph77** [comment]
@gabebw I think your last comment was on my previous commit. I ended up getting rid of @found_hotel.

**Pull request merged**